### PR TITLE
Benchmarks - Integrate cublaslt micro-benchmark

### PR DESCRIPTION
**Pull request by abuccts**
Integrate cublaslt-gemm micro-benchmark #451.

### CODE REVIEW

File: superbench/benchmarks/micro_benchmarks/cublaslt_gemm.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
+# Copyright (c) Microsoft Corporation.
```
Comment (cp5555) [inline superbench/benchmarks/micro_benchmarks/cublaslt_gemm.py:1]: Please add document

File: superbench/benchmarks/micro_benchmarks/cublaslt_gemm.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+"""Module of the cuBLASLt GEMM benchmark."""
+
+import os
+
+from superbench.common.utils import logger
+from superbench.benchmarks import BenchmarkRegistry, Platform, ReturnCode
+from superbench.benchmarks.micro_benchmarks import MicroBenchmarkWithInvoke
+
+
+class LtGemmBenchmark(MicroBenchmarkWithInvoke):
```
Comment (cp5555) [inline superbench/benchmarks/micro_benchmarks/cublaslt_gemm.py:13]: `CublasLtBenchmark`, which is aligned with `CublasBenchmark`

File: superbench/benchmarks/micro_benchmarks/cublaslt_gemm.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                'The result format is invalid - round: {}, benchmark: {}, raw output: {}, message: {}.'.format(
+                    self._curr_run_index, self._name, raw_output, str(e)
+                )
+            )
+            return False
+
+        return True
+
+
+BenchmarkRegistry.register_benchmark('cublaslt-gemm', LtGemmBenchmark, platform=Platform.CUDA)
```
Comment (cp5555) [inline superbench/benchmarks/micro_benchmarks/cublaslt_gemm.py:126]: `CublasLtBenchmark`

File: superbench/benchmarks/micro_benchmarks/cublaslt_gemm.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    def __init__(self, name, parameters=''):
+        """Constructor.
+
+        Args:
+            name (str): benchmark name.
+            parameters (str): benchmark parameters.
+        """
+        super().__init__(name, parameters)
+
+        self._bin_name = 'cublaslt_fp8_gemm'
```
Comment (cp5555) [inline superbench/benchmarks/micro_benchmarks/cublaslt_gemm.py:24]: Maybe we should change bin_name to `cublaslt_benchmark`

Comment (cp5555) [reply superbench/benchmarks/micro_benchmarks/cublaslt_gemm.py:24]: It's not only fp8, also we don't need to highlight gemm

File: superbench/benchmarks/micro_benchmarks/cublaslt_gemm.py (hunk 1 above)
Comment (cp5555) [inline superbench/benchmarks/micro_benchmarks/cublaslt_gemm.py:1]: `cublaslt_function.py`, which is aligned with `cublas_function.py`

**Review by cp5555 — APPROVED** [review]

File: superbench/benchmarks/micro_benchmarks/cublaslt_gemm.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+            required=False,
+            help='Number of warm up steps.',
+        )
+        self._parser.add_argument(
+            '--num_iter',
+            type=int,
+            default=50,
+            required=False,
+            help='Number of iterations to measure.',
+        )
```
Comment (yukirora) [inline superbench/benchmarks/micro_benchmarks/cublaslt_gemm.py:58]: can we change to 'num_steps' to be aligned with other benchmarks

File: superbench/benchmarks/micro_benchmarks/cublaslt_gemm.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+        Return:
+            True if the raw output string is valid and result can be extracted.
+        """
+        self._result.add_raw_data(f'raw_output_{cmd_idx}', raw_output, self._args.log_raw_data)
+
+        try:
+            fields = raw_output.strip().split()
+            if len(fields) != 6 or not all(x.isdigit() for x in fields[:4]):
+                raise ValueError('Invalid result.')
+            self._result.add_result(f'{self._args.in_type}_{"_".join(fields[:3])}', float(fields[-1]))
```
Comment (yukirora) [inline superbench/benchmarks/micro_benchmarks/cublaslt_gemm.py:113]: {self._args.in_type}_{"_".join(fields[:3])}_time or {self._args.in_type}_{"_".join(fields[:3])}_flops ?

**Review by yukirora — APPROVED** [review]

**Pull request merged**